engine/access/rpc/backend/transactions/transactions.go

```diff
@@ -299,9 +303,26 @@ func (t *Transactions) GetTransactionsByBlockID(
 		transactions = append(transactions, collection.Transactions...)
 	}
 
-	transactions = append(transactions, t.systemTx)
+	events, err := t.events.ByBlockID(blockID)
+	if err != nil {
+		return nil, rpc.ConvertStorageError(err)
+	}
+
+	if !t.scheduledCallbacksEnabled {
```

Comment:
after thinking about this more, I don't think we can safely use the flag. If we ever do disable callbacks, we would end up with some blocks including the extra tx and others not. This flag would mean we would return a consistent response for all blocks regardless of what was actually executed.

instead, I think we will just need to lookup the events for each call and resolve the type explicitly.

Reply:
create an issue